pkg/controller/ecr/repositorypolicy/controller.go

```diff
@@ -0,0 +1,214 @@
+	policyData, err := e.formatRepositoryPolicy(cr)
+
+	if err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(err, errGet)
+	}
+
+	if response.PolicyText != nil {
+		cr.Status.AtProvider.PolicyText = *response.PolicyText
+	}
+
+	current := cr.Spec.ForProvider.DeepCopy()
+	ecr.LateInitializeRepositoryPolicy(&cr.Spec.ForProvider, response)
+	if !cmp.Equal(current, &cr.Spec.ForProvider) {
+		if err := e.kube.Update(ctx, cr); err != nil {
+			return managed.ExternalObservation{}, awsclient.Wrap(err, errSpecUpdate)
+		}
+	}
+
+	cr.SetConditions(xpv1.Available())
+
+	return managed.ExternalObservation{
+		ResourceExists:   true,
+		ResourceUpToDate: ecr.IsRepositoryPolicyUpToDate(policyData, &cr.Status.AtProvider.PolicyText),
+	}, nil
+}
+
+func (e *external) Create(ctx context.Context, mgd resource.Managed) (managed.ExternalCreation, error) {
+	cr, ok := mgd.(*v1alpha1.RepositoryPolicy)
+	if !ok {
+		return managed.ExternalCreation{}, errors.New(errUnexpectedObject)
+	}
+
+	cr.Status.SetConditions(xpv1.Creating())
+	if err := e.kube.Status().Update(ctx, cr); err != nil {
+		return managed.ExternalCreation{}, errors.Wrap(err, errStatusUpdate)
+	}
+	policyData, err := e.formatRepositoryPolicy(cr)
+	if err != nil {
+		return managed.ExternalCreation{}, errors.Wrap(err, errCreate)
+	}
+	_, err = e.client.SetRepositoryPolicyRequest(ecr.GenerateSetRepositoryPolicyInput(&cr.Spec.ForProvider, policyData)).Send(ctx)
+	if err != nil {
+		return managed.ExternalCreation{}, awsclient.Wrap(err, errCreate)
+	}
+	return managed.ExternalCreation{}, errors.Wrap(e.kube.Update(ctx, cr), errSpecUpdate)
+}
+
+func (e *external) Update(ctx context.Context, mgd resource.Managed) (managed.ExternalUpdate, error) {
+	cr, ok := mgd.(*v1alpha1.RepositoryPolicy)
+	if !ok {
+		return managed.ExternalUpdate{}, errors.New(errUnexpectedObject)
+	}
+
+	policyData, err := e.formatRepositoryPolicy(cr)
+	if err != nil {
+		return managed.ExternalUpdate{}, errors.Wrap(err, errUpdate)
+	}
+	_, err = e.client.SetRepositoryPolicyRequest(ecr.GenerateSetRepositoryPolicyInput(&cr.Spec.ForProvider, policyData)).Send(ctx)
+	if err != nil {
+		return managed.ExternalUpdate{}, awsclient.Wrap(err, errUpdate)
+	}
+	return managed.ExternalUpdate{}, nil
+}
+
+func (e *external) Delete(ctx context.Context, mgd resource.Managed) error {
+	cr, ok := mgd.(*v1alpha1.RepositoryPolicy)
+	if !ok {
+		return errors.New(errUnexpectedObject)
+	}
+
+	cr.Status.SetConditions(xpv1.Deleting())
+	_, err := e.client.DeleteRepositoryPolicyRequest(&awsecr.DeleteRepositoryPolicyInput{
+		RepositoryName: cr.Spec.ForProvider.RepositoryName,
+		RegistryId:     cr.Spec.ForProvider.RegistryID,
+	}).Send(ctx)
+	if ecr.IsRepoNotFoundErr(err) {
+		return nil
+	}
+	return awsclient.Wrap(resource.Ignore(ecr.IsPolicyNotFoundErr, err), errDelete)
+}
+
+// formatRepositoryPolicy parses and formats the RepositoryPolicy struct
+func (e *external) formatRepositoryPolicy(original *v1alpha1.RepositoryPolicy) (*string, error) {
```

Comment:
```suggestion
func (e *external) rawPolicyData(original *v1alpha1.RepositoryPolicy) (string, error) {
```

nitpick: returning `string` could make it easier to work with. We'd use `awsclients.String(ref)` as usual whenever we send it to AWS. And `formatRepositoryPolicy` feels like there is an in-place change but what we do is mostly conversion from one format to another.

Reply:
Also what do you think about moving this to `pkg/clients/ecr/repository_policy.go`?

Reply:
kk and i'll go ahead and move it as well